src/openrct2-ui/interface/Graph.cpp

```diff
@@ -182,10 +183,132 @@ static void graph_draw_line_b_money32(
     }
 }
 
+struct FinancialTooltipInfo
+{
+    const CoordsXY coords;
+    const money32 money{};
+};
+
+struct Rect
+{
+    const ScreenCoordsXY LeftTop;
+    const ScreenCoordsXY RightBottom;
+
+    int32_t GetLeft() const
+    {
+        return LeftTop.x;
+    }
+    int32_t GetTop() const
+    {
+        return LeftTop.y;
+    }
+    int32_t GetRight() const
+    {
+        return RightBottom.x;
+    }
+    int32_t GetBottom() const
+    {
+        return RightBottom.y;
+    }
+    int32_t GetWidth() const
+    {
+        return RightBottom.x - LeftTop.x;
+    }
+    int32_t GetHeight() const
+    {
+        return RightBottom.y - LeftTop.y;
+    }
+    bool Contains(const CoordsXY& coords) const
```

Comment:
This is dangerous and shouldn't be performed. CoordsXY represent map coordinates this Rect represents screen coordinates they should never be compared directly (apart from on the map window but this isn't used for the map window)